src/logger.c

```diff
@@ -0,0 +1,127 @@
+#include "logger.h"
+
+#include "constants.h"
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+LoggerData g_logger;
+
+#ifndef min
```

Comment:
I thought IRC said you didn't need this as it exists in math.h? If you do need it, can you move it out to another file so it can be used elsewhere in the code.